Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -715,14 +721,56 @@ private class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
       let visitedNode = self.visit(declSyntax)
       skipVisitAnyHandling.remove(Syntax(declSyntax))
 
-      let attributesToRemove = self.macroAttributes(attachedTo: visitedNode).map(\.attributeNode)
-
-      return AttributeRemover(removingWhere: { attributesToRemove.contains($0) }).rewrite(visitedNode)
+      let attributesToRemove = self.macroAttributes(attachedTo: visitedNode)
+      attributesToRemove.forEach { (attribute, spec) in
```

Comment:
I generally prefer `for … in … ` over `.forEach`

```suggestion
      for (attribute, spec) in attributesToRemove {
```